pytorch_toolkit/formula_recognition/tools/train.py

```diff
@@ -0,0 +1,252 @@
+        self.val_results_path = os.path.join(self.save_dir, "val_results")
+        self.step = 0
+        self.global_step = 0
+        self._test_steps = config.get('_test_steps', 1e18)
+        self.epoch = 1
+        self.best_val_loss = 1e18
+        self.best_val_accuracy = 0.0
+        self.best_val_loss_test = 1e18
+        self.best_val_accuracy_test = 0.0
+        self.print_freq = config.get('print_freq', 16)
+        self.save_freq = config.get('save_freq', 2000)
+        self.val_freq = config.get('val_freq', 5000)
+        self.logs_path = os.path.join(self.work_dir, 'logs')
+        self.writer = SummaryWriter(self.logs_path)
+        self.writer.add_text("General info", pformat(config))
+        self.create_dirs()
+        self.load_dataset()
+        self.model = Im2latexModel(config.get('backbone_type'), config.get(
+            'backbone_config'), len(self.vocab), config.get('head'))
+        if self.model_path is not None:
+            self.model.load_weights(self.model_path, old_model=config.get("old_model"))
+
+        self.optimizer = getattr(optim, config.get('optimizer', "Adam"))(self.model.parameters(), self.learing_rate)
+        self.lr_scheduler = ReduceLROnPlateau(self.optimizer)
+        self.device = config.get('device', 'cpu')
+        self.model = self.model.to(self.device)
+        self.time = get_timestamp()
+
+    def create_dirs(self):
+        if not os.path.exists(self.logs_path):
+            os.makedirs(self.logs_path)
+        if not os.path.exists(self.save_dir):
+            os.makedirs(self.save_dir)
+        if not os.path.exists(self.val_results_path):
+            os.makedirs(self.val_results_path)
+
+    def load_dataset(self):
+        train_dataset = ConcatDataset(
+            [Im2LatexDataset(train_s, 'train') for train_s in self.train_paths]
+        )
+        train_sampler = BatchRandomSampler(
+            dataset=train_dataset, batch_size=self.config.get('batch_size', 4))
+        batch_transform_train = create_list_of_transforms(self.val_transforms_list)
+        self.train_loader = DataLoader(
+            train_dataset,
+            batch_sampler=train_sampler,
+            collate_fn=partial(collate_fn, self.vocab.sign2id,
+                               batch_transform=batch_transform_train),
+            num_workers=os.cpu_count())
+
+        val_dataset = Im2LatexDataset(self.val_path, 'validate')
+        val_sampler = BatchRandomSampler(dataset=val_dataset, batch_size=1)
+        batch_transform_val = create_list_of_transforms(self.val_transforms_list)
+        self.val_loader = DataLoader(
+            val_dataset,
+            batch_sampler=val_sampler,
+            collate_fn=partial(collate_fn, self.vocab.sign2id,
+                               batch_transform=batch_transform_val),
+            num_workers=os.cpu_count())
+
+    def train(self):
+
+        print("Created logs folder: {}".format(self.logs_path))
```

Comment:
Why it is printed here? It is confusing!    
I;d move the print to the function `create_dirs`

Reply:
Moved and added log message about the creation of the other folders (save dir, val results dir)